components/Checkbox/Checkbox.tsx

```diff
@@ -1,31 +1,47 @@
+import { useState } from "react";
+
 export interface CheckboxProps
   extends React.InputHTMLAttributes<HTMLInputElement> {
   id: string;
   label: string;
 }
 
 export const Checkbox = ({ label, id, ...props }: CheckboxProps) => {
+  const [focus, setFocus] = useState<string>("");
+
+  const containerStyle = "flex max-w-[13rem] items-center rounded p-1";
+
   const inputStyle =
     "peer appearance-none w-5 min-w-[1.25rem] h-5 border rounded-[2px] border-blue cursor-pointer checked:border-orange checked:bg-orange hover:enabled:border-2 focus:outline-none disabled:cursor-not-allowed disabled:border-gray-dark disabled:bg-gray-medium";
 
+  const labelStyle =
+    "text-blue text-sm font-outfit ml-2 peer-checked:text-orange font-medium peer-hover:font-bold peer-disabled:text-gray-dark peer-disabled:font-medium";
+
   const formattedId = id.toLowerCase().split(" ").join("");
 
   if (!label || label.trim().length === 0) {
     return null;
   }
 
+  const handlerOnClick = () => {
+    setFocus("");
+  };
+
+  const handlerOnFocus = () => {
+    setFocus("focus-within:shadow-2md");
+  };
+
   return (
-    <div className="flex max-w-[13rem] items-center focus-within:shadow-2md p-1 rounded">
+    <div className={`${containerStyle} ${focus}`}>
       <input
         id={formattedId}
         type="checkbox"
         className={inputStyle}
         {...props}
+        onClick={handlerOnClick}
+        onFocus={handlerOnFocus}
```

Comment:
Podemos hacer esto sin eventos? especificamente sin `useState`, queremos dejar los componentes de forma *stateless*

Probablemente [con esto](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-sibling-state) se pueda llevar a lo que queremos

Reply:
Bien @aandoro, como te decia.. como `[&:has(input:focus-visible)]:shadow-2md` no funcionaria funciona para Firefox (por el `:has()`)

Otra cosa que se me ocurre con `peer` es agregar otro elemento `div` justo despues del `label`, hacer que tome el tamanio del `div` parent (o container) y aplicarle el `shadow-2md` y `focus-visible`